lib/dal/src/secret.rs

```diff
@@ -291,6 +326,69 @@ impl Secret {
         self.key
     }
 
+    /// Attach a [`Secret`] to a given [`AttributeValue`] corresponding to a
+    /// "/root/secrets/\<secret\>" [`Prop`](crate::Prop).
+    ///
+    /// - If a [`Secret`] has already been attached to the given [`AttributeValue`], the existing
+    ///   attachment will be replaced with a new one.
+    /// - If no [`Secret`] is provided, then the [`AttributeValue`] will be restored to its default
+    ///   state.
+    ///
+    /// This method will enqueue
+    /// [`DependentValuesUpdate`](crate::job::definition::DependentValuesUpdate).
+    pub async fn attach_for_attribute_value(
+        ctx: &DalContext,
+        attribute_value_id: AttributeValueId,
+        secret_id: Option<SecretId>,
+    ) -> SecretResult<()> {
+        // First, check if the caller would like to unset the secret.
+        let secret_id = match secret_id {
+            Some(provided_secret_id) => provided_secret_id,
+            None => {
+                AttributeValue::use_default_prototype(ctx, attribute_value_id).await?;
```

Comment:
I'm still thinking through whether or not the user would want a scenario where they'd want to override the default "I get my secret from an input socket" prototype, but they would still want it to "unset". We may actually just want to unset the value and keep the prototype.